L1Trigger/L1TGlobal/plugins/L1TGlobalProducer.cc

```diff
@@ -71,6 +74,9 @@ void L1TGlobalProducer::fillDescriptions(edm::ConfigurationDescriptions& descrip
   // switch for muon showers in Run-3
   desc.add<bool>("useMuonShowers", false);
 
+  //switch for saving AXO score
+  desc.add<bool>("saveAxoScore", true);
```

Comment:
Hi Marino, we thought it would be ok to set this to true by default even if the firmware side might never have this feature added to make it easier for people to use this score. Is there a downside to always producing this score?

Reply:
Hi Efe, I'm thinking (as usual) about the L1T emulator instance running inside the HLT. Since nothing in the HLT would need this score, I would keep this `false` inside HLT. If the default is `false`, this will happen automatically. If the default is `true`, then we need a JIRA ticket to set this to `false` in the HLT menu. That's at least my take. @cms-sw/hlt-l2 can take it from here.

Reply:
I think our main motivation was that with this the GT emulator would produce the score and then this would be added to the event content. Otherwise the score would not be produced and not saved in the event content.

Reply:
(and the idea is to develop some DQM monitoring based on the saved score! cc @thesps )

Reply:
Added to the EventContent where? Of L1T or HLT as the emulator is run in HLT? I suggest to make a CMSHLT Jira ticket to cover the required changes.

Reply:
we need to improve a bit the communication chain. Was this ever discussed with TSG? At least it's news to me.

Reply:
I echo Martin's question.

> I think our main motivation was that with this the GT emulator would produce the score and then this would be added to the event content. Otherwise the score would not be produced and not saved in the event content.

Already now, w do not save individual L1T objects/values in the HLT EventContent. We instead save the RAW data of the L1uGT FED (e.g. in NanoDST), or even the RAW of all FEDs altogether (e.g. DQM stream). All this should be sufficient to recreate the AXO score by running the L1uGT emulator "offline", either in the online-dqm client or in offline workflows. I'm not seeing the need to change any of the HLT EventContents.

> we need to improve a bit the communication chain. Was this ever discussed with TSG? At least it's news to me.

Agreed. It was not discussed with me, fwiw.

Reply:
This is exactly right, @eyigitba

Reply:
> My understanding was that this feature was developed by AXO developers/users to have access to AXO score in offline event content.

So, could we set the default to `False` so that the instance of the L1uGT emulator running inside HLT automatically gets the correct value for HLT, and let L1T update the instance of the L1uGT emulator which is supposed to have `saveAxoScore=True` to fill the "offline event content" ? (in quotes because I do not know exactly which EventContent this is)

Reply:
What I meant with "offline event content" was that developers wanted to have this in MC production GSDR samples to then propagate this to L1Ntuples or L1Nano seemlessly (maybe I was wrong about this).

Reply:
Why is it a problem to have it set to true as default for HLT? Wouldn't that just mean that the score is calculated inside the L1TGlobalProducer but not used? The only place it is saved is within the L1TNtuples, which is enabled/disabled with a token for example defined [here](https://github.com/cms-sw/cmssw/blob/6a131029e88625000a80f6618e4a173d325fffc0/L1Trigger/L1TNtuples/python/l1AXOTree_cfi.py#L4) and enabled [here](https://github.com/cms-sw/cmssw/blob/6a131029e88625000a80f6618e4a173d325fffc0/L1Trigger/L1TNtuples/python/L1NtupleEMU_cff.py#L65):  If the token is not listed the score is not saved to the ntuple.

Reply:
> Wouldn't that just mean that the score is calculated inside the L1TGlobalProducer but not used?

Yes, as far as I understand.

> Why is it a problem to have it set to true as default for HLT?

The logic is the opposite one, imho. "Why would this need to run at HLT ?", and the answer is apparently "no reason".

Reply:
I can rename the variable to produceAXOScore if you like, as it doesn't make a difference. However I will defer to @aloeliger and @eyigitba with regards to setting the default to false. I think it would be easier to have it false to avoid error when creating ntuples/running offline dqm and because it makes no difference to HLT except the saving of a variable in the GlobalProducer.

Reply:
If the default is not set to `False`, and this is backported to 14_0_X, this will require the parsing a new 14_0_X release into ConfDB in order to (a) expose the new parameter in the HLT configurations, and (b) set it to `False` in the HLT menus (because it does not need to be `True` in the instance of the L1T emulator running online inside the HLT menu).

For the name, yes, please use `produceAXOL1TLScore` as it is clearer, imho.

Reply:
Pardon my confusion, but why would it need to be set to false in HLT configurations? What would happen if it is set to true? And would the parameter need to be exposed only if it is set to true, or also when set to false?

Reply:
See https://github.com/cms-sw/cmssw/pull/45227#discussion_r1640889729.

> Since nothing in the HLT would need this score, I would keep this false inside HLT. If the default is false, this will happen automatically. If the default is true, then we need a JIRA ticket to set this to false in the HLT menu.

> And would the parameter need to be exposed only if it is set to true, or also when set to false?

If you leave it to false, we'll just take it from `fillDescriptions`, no need to change it.